pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -526,94 +532,139 @@ func (r *ReconcileCredentialsRequest) Reconcile(ctx context.Context, request rec
 	} else {
 		crSecretExists = true
 	}
+	if stsFeatureGateEnabled && stsDetected {
+		// create time-based tokens based on settings in CredentialsRequests
+		logger.Infof("timed token access cluster detected: %t, so not trying to provision with root secret",
+			stsDetected)
+		credsExists, err := r.Actuator.Exists(context.TODO(), cr)
+		if err != nil {
+			logger.Errorf("error checking whether credentials already exists: %v", err)
+			return reconcile.Result{}, err
+		}
 
-	credentialsRootSecret, err := r.Actuator.GetCredentialsRootSecret(context.TODO(), cr)
-	if err != nil {
-		log.WithError(err).Debug("error retrieving cloud credentials secret, admin can remove root credentials in mint mode")
-	}
-	cloudCredsSecretUpdated := credentialsRootSecret != nil && credentialsRootSecret.ResourceVersion != cr.Status.LastSyncCloudCredsSecretResourceVersion
-	isStale := cr.Generation != cr.Status.LastSyncGeneration
-	hasRecentlySynced := cr.Status.LastSyncTimestamp != nil && cr.Status.LastSyncTimestamp.Add(syncPeriod).After(time.Now())
-	hasActiveFailureConditions := checkForFailureConditions(cr)
+		var syncErr error
+		if !credsExists {
+			syncErr = r.Actuator.Create(context.TODO(), cr)
```

Comment:
nit: here and elsewhere - you have a context available to you in `Reconcile()` - no need for TODO

Reply:
yep, eliminated `TODO()` usage everywhere in this file